File: leetcode/solutions/0454.4SumII/fourSumCount_test.go
package solution

import (
	"testing"
)

func Test_0454_4SumII(t *testing.T) {
	t.Parallel()

	type in struct {
		arg1 []int
		arg2 []int
		arg3 []int
		arg4 []int
	}
	tests := []struct {
		in  in
		out int
	}{
		{
			in{
				[]int{1, 2},
				[]int{-2, -1},
				[]int{-1, 2},
				[]int{0, 2},
			},
			2,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := fourSumCount(tt.in.arg1, tt.in.arg2, tt.in.arg3, tt.in.arg4)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0454.4SumII/fourSumCount_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import fourSumCount from "./fourSumCount.ts";

Deno.test("0454. 4Sum II", () => {
  type test = [[number[], number[], number[], number[]], number];
  const tests: test[] = [
    [
      [
        [1, 2],
        [-2, -1],
        [-1, 2],
        [0, 2],
      ],
      2,
    ],
  ];

  for (const [args, want] of tests) {
    assertEquals(fourSumCount(...args), want);
  }
});
